### DISCUSSION TITLE
E-ID standards, OSS and interop

### DISCUSSION THREAD

**Discussion by damienbod** [e-id-admin/governance-sounding-board#49] [General]
I've implemented a few different SSI solutions using different SSI vendors and I am wondering if interop will work with E-ID.  I cannot test against the sandbox solutions because the sandbox uses Hyperleger Aries and none of the SSI solutions I implemented supported this. Every vendor implements against standards, just not the same ones...

From what I understand these 2 standards will be required (OpenID for Verifiable Presentations and OpenID4VCI ) for eidas and so the Schweiz E-ID. 

1) Do you have a scheme definition for the E-ID? (WIP in OSS would be great) 
2) Non of the code is open source, why do this behind closes doors? If the code was public, you get free security testing.
3) When defining schemes, is there a body or somewhere we can just re-use the same ones?
4) Do you have a plan for the standards which will be supported?  Have you defined any yet? If you want to have this completed by 2025, this should be clear now.

Kind regards Damien

**Comment by christianheimann** [comment]
Great to have your voice and experiences still on board, and thank you for your questions and interest.
1. Actually there is not yet a technical schema definition. In the actual draft of the law there is a list of attributes which will take place in the E-ID. For the draft see here: [Vernehmlassungsunterlagen (admin.ch)](https://www.eid.admin.ch/eid/de/home/dokumentation/vernehmlassungsunterlagen.html)
2. At the moment, we are in a PoC phase, and we won’t go productive with this code. Therefore this code is actually not public. For the final-implementation this question has to be answered in the law if the code is open. For technical manner it would be good to have the code public.
3. The reuse of schemes depends on governance decisions as well as on the technical stack, that will be selected. But we're not there yet. As not every stack needs schemas as info on the base registry, it also could simply be defined in a kind of verifiable credentials catalog every one can point to.
4. The technical evaluation is ongoing at the moment and will be presented regularly at the participation meetings; and we are aware of the tough timeline. 

Hope this answers give you a direction. If you have any experiences from your implementation you want to share, your further comments are appreciated!

**Comment by damienbod** [comment]
Hi  @christianheimann thanks, I am really interested in this.

FYI

The EUDI Wallet Technical Specifications are very interesting, they almost make the exact same same thing and it is EUDI conform. The PID Data Model and the documentation is really interesting. The swiss project could do public documentation like this as well? :)

https://github.com/italia/eudi-wallet-it-docs/tree/versione-corrente/docs/en

Used Standards:

## Draft: OAuth 2.0 Attestation-Based Client Authentication

https://datatracker.ietf.org/doc/html/draft-looker-oauth-attestation-based-client-auth-00

## Draft: OpenID for Verifiable Presentations

https://openid.net/specs/openid-4-verifiable-presentations-1_0.html

## RFC 9449 OAuth 2.0 Demonstrating Proof-of-Possession at the Application Layer (DPoP)

https://www.rfc-editor.org/rfc/rfc9449.html

## Draft: OpenID for Verifiable Credential Issuance

https://openid.bitbucket.io/connect/openid-4-verifiable-credential-issuance-1_0.html

## Draft: OpenID Connect for Identity Assurance 1.0

https://openid.net/specs/openid-connect-4-identity-assurance-1_0-13.html

## Draft: SD-JWT-based Verifiable Credentials (SD-JWT VC)

https://www.ietf.org/archive/id/draft-terbu-oauth-sd-jwt-vc-00.html
